palace/fem/multigrid.hpp

```diff
@@ -70,17 +67,15 @@ std::vector<std::unique_ptr<FECollection>> inline ConstructFECollections(
     }
   }
   std::reverse(fecs.begin(), fecs.end());
+
   return fecs;
 }
 
-// Construct a hierarchy of finite element spaces given a sequence of meshes and
-// finite element collections. Dirichlet boundary conditions are additionally
-// marked.
+// Construct a hierarchy of finite element spaces given a sequence of meshes and finite
+// element collections. Additionally, Dirichlet boundary conditions are marked.
 template <typename FECollection>
-inline std::unique_ptr<mfem::ParFiniteElementSpaceHierarchy>
-ConstructFiniteElementSpaceHierarchy(
-    int mg_max_levels, bool mg_legacy_transfer, int pa_order_threshold,
-    bool pa_discrete_interp, const std::vector<std::unique_ptr<mfem::ParMesh>> &mesh,
+inline FiniteElementSpaceHierarchy ConstructFiniteElementSpaceHierarchy(
+    int mg_max_levels, const std::vector<std::unique_ptr<mfem::ParMesh>> &mesh,
```

Comment:
Given we've now taken control of space hierarchies, these should probably be constructors for the relevant classes rather than free functions.

Reply:
As discussed offline, I prefer to keep the constructors for `FiniteElementSpaceHierarchy` simple and include these helpers as free functions.